src/ripple/nodestore/ShardSizeTuning.md

```diff
@@ -0,0 +1,83 @@
+# Shard Size Tuning
+
+The purpose of this document is to compare shard size of 8K and 16K ledgers.
+
+## Methodology
+
+One can see visually from a block explorer that typical mainnet ledger
+consists of about 30 offer transactions issued by about 8 different accounts,
+and several transactions of other types. To simulate this situation and
+similar situations we had constructed deterministic shards of differenet sizes,
+with different number of offer transactions per ledger and different number
+of accounts issuing these offers.
+
+Number of ledgers per shard starts from 256 to 16K with doubling the size on
+each step. We considered the following numbers of offers per ledger:
+0, 1, 5, 10 and 30. Also we considered case of 1 and 8 accounts issuing offers.
+For each constructed deterministic shard we counted its size. Finally we
+compared doubled size of the shard with N ledgers and the size of a shard with
+2*N ledgers where othere parameters such as number of offers and accounts are
+the same.
+
+
+## Raw results table
+
+All sizes of constructed shards are shown in the following table.
+Rows corresponds to shard sizes (S) counted in ledgers, columns corresponds
+to numbers of offers (O) per ledger. In each cell there are two numbers:
+first number corresponds to the case of 1 account issuing offers, the second
+number corrspoonds to 8 accounts. Each number is a size of the shard with
+given parameters measured in megabytes.
+
+|S\O|0|1|5|10|30|
+|---|---|---|---|---|---|
+|256|2.2/2.2|3.4/3.3|5.3/7.3|7.7/10.9|17.1/21.9|
+|512|4.4/4.5|7.0/7.0|11.2/15.6|16.4/23.7|36.9/47.9|
+|1K|8.9/9.0|14.7/14.6|23.7/33.6|35.0/51.0|78.2/ 102.9|
+|2K|17.8/18.0|30.5/30.7|50.4/72.2|74.3/ 111.9|166.2/ 221.0|
+|4K|35.5/35.9|63.6/64.2|106.2/ 154.8|156.1/ 238.7|354.7/ 476.0|
+|8K|71.1/71.9|133.4/ 134.5|222.2/ 328.1|329.1/ 511.2|754.6/ 1021.0|
+|16K|142.3/ 143.9|279/9 280.8|465.7/ 698.1|696.4/ 1094.2|1590.5/ 2166.6|
+
+## Preliminary conclusion
+
+If one compares a doubled size of shard with N ledgers and a size of shard
+with 2*N ledgers anythere in the above table than the conlusion will be that
+the second number is greater. For example, the following table shows the
+percentage in which the second number is greater for the most interesting case
+of 30 offers per ledger. The first row corresponds to the case of 1 account
+issuing offers, and the second row corresponds to the case of 8 issuing
+accounts.
+
+|A\N|256|512|1K|2K|4K|8K|
+|---|---|---|---|---|---|---|
+|1|8%|6%|6%|6%|7%|6%|5%|
+|8|9%|7%|7%|8%|6%|7%|6%|
+
+The common conclusion in this model is that if one doubled shard size then
+the total disk space utilized will raise by 5-9%.
+
+## Adding accounts into consideration
+
+Previous model does not take into account that there are large number of
+XRP accounts in the mainnet, and each shard should contain information
+about each of these accounts. As of January 2020, there was about 1.9 millions
```

Comment:
Change to _"... there were about 1.9 million ..."_.

Reply:
Done.